tutorials/using-stackdriver-uptime-checks-for-scheduling-cloud-functions/index.md

```diff
@@ -0,0 +1,518 @@
+<pre>
+const Datastore = require('@google-cloud/datastore');
+const GCS = require('@google-cloud/storage');
+const Request = require('request');
+const BUCKET = '<b>[CRON_BUCKET]</b>'; // Replace with your bucket.
+const TRIGGER_LIST = 'trigger.json';
+
+/**
+ * Cloud Function.
+ *
+ * @param {Object} req Cloud Function request context.
+ * @param {Object} res Cloud Function response context.
+ */
+exports.cronMaster = function cronMaster(req, res) {
+    checkTimestamp()
+        .then(() => readTriggerList())
+        .then((triggerList) => runTriggerList(res, triggerList))
+        .catch(() => res.status(200).end());
+};
+</pre>
+
+The `cronMaster` function begins by calling the `checkTimestamp` helper function to see if it’s time to process the trigger list. If so, `cronMaster` calls the `readTriggerList` helper function to parse the contents of the JSON-formatted trigger list in Cloud Storage into a `triggerList` object. Finally, it passes the `triggerList` object to the `runTriggerList` helper function to trigger the scheduled Cloud Functions.
+
+Each of these helper functions return a "[`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)" that only resolves for this [happy path](https://en.wikipedia.org/wiki/Happy_path), making it easy to implement the logic by chaining them together using `then` handlers. In all other cases -- if it’s not time to process the trigger list, or if an exception is raised along the way -- the exception handler is called.  In this implementation, the exception handler does nothing except return a `200` HTTP status code.
+
+### Understanding `checkTimestamp`
+
+The `checkTimestamp` function works by reading a `Cron` entity from Cloud Datastore, that has a `timestamp` field indicating the last time the trigger list was processed.
+<pre>
+function checkTimestamp() {
+    var now = new Date().getTime() / 1000;
+
+    var datastore = Datastore();  // Authenticating on a global basis.
+    var key = datastore.key(['Cron', '1 Minute']);
+    const transaction = datastore.transaction();
+
+    return transaction.run()
+        .then(() => transaction.get(key))
+        .then((results) => {
+            var entity = results[0];
+            if (!entity || now - entity.timestamp >= 60) {
+                transaction.save({key: key, data: {timestamp: now}});
+                return transaction.commit();
+            } else {
+                throw 'It is not time yet';
+            }
+        }).catch((e) => {
+            transaction.rollback();
+            throw e;
+        });
+}
+</pre>
+
+If enough time has elapsed, 60 seconds in this example, or if the entity is not found, which is the case the first time this function is executed, `checkTimestamp` does the following:
```

Comment:
If enough time has elapsed  (60 seconds in this example), or if the entity is not found